src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,68 +3,61 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
-    private static final int KEY_IN_ARRAY = 0;
-    private static final int VALUE_IN_ARRAY = 1;
-    private static final int KEY_VALUE_PAIR_LENGTH = 2;
-
-    private int arrayLengthCounter = 0;
-    private Object[][] newArray = new Object[0][0];
+    private int arrayLengthCounter;
+    private Object[] keyArray = new Object[0];
+    private Object[] valueArray = new Object[0];
```

Comment:
it's better to set non-constant fields in the constructor
```suggestion
    private Object[] keys;
    private Object[] values;
```